filter_plugins/cisco_nxos.py

```diff
@@ -0,0 +1,370 @@
+
+    # Remove lines that will be filled in by this plugin
+    delete_section = False
+    found_vlanrange_line = False
+    for line in config:
+        if line.startswith("!"):
+            # skip commands
+            continue
+
+        if line == "" and delete_section:
+            # Skip empty lines if deleting
+            continue
+
+        if not line.startswith(" "):
+            # This is a new section
+            delete_section = False
+
+        if line.startswith("interface breakout "):
+            # Existing breakout config, get port
+            match = re.search(r"port (\d+)", line)
+            if match:
+                port_num = match.group(1)
+                object_name = f"Ethernet1/{port_num}"
+                if object_name in interfaces:
+                    continue
+
+        if line.startswith("interface "):
+            # This is an interface config
+
+            # Find the object name
+            object_name = line.removeprefix("interface ")
+            if not object_name.startswith("Ethernet") and not object_name.startswith("port-channel"):
+                delete_section = False;
+                new_config.append(line)
+                continue
+
+            # Check if object is configured
+            if object_name.count("/") > 1:
+                # this is a breakout
+                object_name = object_name[:object_name.rfind("/")]
+
+            if object_name in interfaces:
+                if "managed" in interfaces[object_name] and interfaces[object_name]["managed"]:
+                    delete_section = False
+                    new_config.append(line)
+                    continue
+                else:
+                    delete_section = True
+                    continue
+            else:
+                new_config.append(line)
+                delete_section = False
+                continue
+
+        if line.startswith("vlan "):
+            if not found_vlanrange_line:
+                # Skip vlan range line
+                found_vlanrange_line = True
+                continue
+
+            # This is a VLAN config
+            object_name = int(line.removeprefix("vlan "))
+            if object_name in vlans:
+                if "managed" in vlans[object_name] and vlans[object_name]["managed"]:
+                    delete_section = False
+                    new_config.append(line)
+                    continue
+                else:
+                    delete_section = True
+                    continue
+            else:
+                delete_section = True
+                continue
+
+        if not delete_section:
+            new_config.append(line)
+
+    return new_config
+
+def generateVLANConfig(vlans):
+    config = []
+    for id,fields in vlans.items():
+        if "managed" in fields and fields["managed"]:
+            continue
+
+        config.append("")  # Newline
+        config.append(f"vlan {str(id)}")
+        config.append(f"  name {fields["name"]}")
```

Comment:
syntax error here because you are using the same quotes.
```
    config.append(f" name {fields["name"]}")
                                   ^
SyntaxError: f-string: unmatched '['
```

I would create a multi-line string instead of calling config.append mutliple lines.

Reply:
I think it's okay to use double quotes like this because python actually parses the {} separately. It works fine like that when I run it.

I will make it multiline

Reply:
Well, actually if you got a syntax error I'm not sure, maybe it's just the way ansible runs filter plugins that works better? I'll just make them single quotes

Reply:
I am kind of surprised that this works. Anyway, it fails for me.
```
➜  ~ cat test.py
fields = {}
config = {}
id = "23"
fields["name"] = "foo"

config.append("")  # Newline
config.append(f"vlan {str(id)}")
config.append(f"  name {fields["name"]}")
config.append(f" name {fields["name"]}")

print(lines)

➜  ~ python test.py
  File "/Users/naved/test.py", line 8
    config.append(f"  name {fields["name"]}")
                                    ^
SyntaxError: f-string: unmatched '['
```

Reply:
That is a python version issue. The syntax is correct with current (3.12) python. Not sure exactly when it became valid.